connect/runtime/src/test/java/org/apache/kafka/connect/integration/ErrantRecordSinkConnector.java

```diff
@@ -54,7 +65,16 @@ public void put(Collection<SinkRecord> records) {
                     .computeIfAbsent(rec.topic(), v -> new HashMap<>())
                     .computeIfAbsent(rec.kafkaPartition(), v -> new TopicPartition(rec.topic(), rec.kafkaPartition()));
                 committedOffsets.put(tp, committedOffsets.getOrDefault(tp, 0) + 1);
-                reporter.report(rec, new Throwable());
+                Throwable error = new Throwable();
+                // Test synchronous and asynchronous reporting, allowing for re-ordering the errant reports
```

Comment:
We don't have any corresponding verification that this behavior is handled correctly. If there's an easy, lightweight way to add that, it'd be nice, but it's not worth blocking on if it's too cumbersome.

Reply:
What verification are you thinking about? Currently we're asserting that at least N errors are reported in total after delivering N records.

Reply:
Ah, that's true. Can we also verify that the reported records don't include any duplicates? Would add at least a little more safety here in case refactoring caused successive calls to `ErrantRecordReporter::report` to overwrite each other.